pkg/dogstatsd/listeners/packets_buffer.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package listeners
+
+import (
+	"sync"
+	"time"
+)
+
+// packetsBuffer is a buffer of packets that will automatically flush to configurable channel
+// when it is full or after a configurable duration
+type packetsBuffer struct {
+	packets       Packets
+	flushTimer    *time.Ticker
+	bufferSize    uint
+	outputChannel chan Packets
+	closeChannel  chan struct{}
+	m             sync.Mutex
+}
+
+func newPacketsBuffer(bufferSize uint, flushTimer time.Duration, outputChannel chan Packets) *packetsBuffer {
+	pb := &packetsBuffer{
+		bufferSize:    bufferSize,
+		flushTimer:    time.NewTicker(flushTimer),
+		outputChannel: outputChannel,
+		packets:       make(Packets, 0, bufferSize),
+		closeChannel:  make(chan struct{}),
+	}
+	go pb.flushLoop()
+	return pb
+}
+
+func (pb *packetsBuffer) flushLoop() {
+	for {
+		select {
+		case <-pb.flushTimer.C:
+			pb.m.Lock()
+			pb.flush()
+			pb.m.Unlock()
+		case <-pb.closeChannel:
+			return
+		}
+	}
+}
+
+func (pb *packetsBuffer) append(packet *Packet) {
+	pb.m.Lock()
+	defer pb.m.Unlock()
+	pb.packets = append(pb.packets, packet)
+	if uint(len(pb.packets)) == pb.bufferSize {
```

Comment:
I don't see any reason for now to do this (the lock is here and everything), but, I think that's would be better to do a `>=` here ^^